test_runner/regress/test_sharding.py

```diff
@@ -1703,20 +1753,16 @@ def test_back_pressure_per_shard(neon_env_builder: NeonEnvBuilder):
             "neon.max_cluster_size = 10GB",
         ],
     )
-    endpoint.respec(skip_pg_catalog_updates=False)  # Needed for databricks_system to get created.
+    endpoint.respec(skip_pg_catalog_updates=False)
     endpoint.start()
 
-    # generate 20MB of data
+    # generate 10MB of data
```

Comment:
This was causing me to hit statement_timeout issues, so I changed it to 10MB. I will backport the fix to Hadron. @HaoyuHuang to review this PR.

Reply:
20 MB should work. The test set 15 MB and 4 shards. so PG should hit backpressure on 60 MB. 

Do you have this code in?
int tenant_max_replication_apply_lag = num_shards * max_replication_apply_lag;